### DISCUSSION TITLE
Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }

### DISCUSSION THREAD

**Discussion by l1x** [bottlerocket-os/bottlerocket#1475] [Q&A] [answered]
**Platform I'm building on:**

ARM64 / Raspberry Pi

**What I expected to happen:**

Build succeed.

**What actually happened:**

```
> cargo make -e BUILDSYS_VARIANT=parami-dev
[cargo-make] INFO - cargo make 0.32.15
[cargo-make] INFO - Project: parami-dev
[cargo-make] INFO - Build File: Makefile.toml
[cargo-make] INFO - Task: default
[cargo-make] INFO - Profile: development
[cargo-make] INFO - Execute Command: "cargo" "fmt"
[cargo-make] INFO - Execute Command: "cargo" "build" "--all-features"
   Compiling parami-dev v0.1.0 (/home/l1x/code/bottlerocket/variants/parami-dev)
error: failed to run custom build command for `parami-dev v0.1.0 (/home/l1x/code/bottlerocket/variants/parami-dev)`

Caused by:
  process didn't exit successfully: `/home/l1x/code/bottlerocket/variants/parami-dev/target/debug/build/parami-dev-8f58e588b24b9bbb/build-script-build` (exit code: 1)
  --- stderr
  Error: Os { code: 2, kind: NotFound, message: "No such file or directory" }
[cargo-make] ERROR - Error while executing command, exit code: 101
[cargo-make] WARN - Build Failed.
```

**How to reproduce the problem:**

Git clone repo and try to build.

I am just trying to understand how to debug this. I was trying to look for the error but I could not identify where it is coming from.

**Comment by sanu11** [comment, marked as answer]
Hello @l1x 

Thanks for reaching out! The error means the variant you're specifying with `BUILDSYS_VARIANT`  ("parami-dev" in this case)  does not exist. You can look for the valid variants [here](https://github.com/bottlerocket-os/bottlerocket#variants). If you are looking to create your own variant, you can follow the instruction [here](https://github.com/bottlerocket-os/bottlerocket/blob/develop/variants/README.md#development). Let us know if you have any further questions.

**Comment by l1x** [comment]
Makes sense, the  cargo make command has to be executed from the project root and it works. Thank you!
